fuel-core/src/database.rs

```diff
@@ -248,3 +260,113 @@ impl InterpreterStorage for Database {
         Ok(block.headers.producer)
     }
 }
+
+#[async_trait]
+impl RelayerDb for Database {
+    /// get validator set for current eth height
+    async fn current_validator_set(&self) -> HashMap<Address, u64> {
+        struct WrapAddress(pub Address);
+        impl From<Vec<u8>> for WrapAddress {
+            fn from(i: Vec<u8>) -> Self {
+                Self(Address::try_from(i.as_ref()).unwrap())
+            }
+        }
+        let mut out = HashMap::new();
+        for diff in self.iter_all::<WrapAddress, u64>(columns::VALIDATOR_SET, None, None, None) {
+            match diff {
+                Ok((address, stake)) => {
+                    out.insert(address.0, stake);
+                }
+                Err(_) => return out,
```

Comment:
Shouldn't we return the actual database error instead of swallowing and returning a default response? At the very least the db error should be logged for troubleshooting purposes.

Reply:
looks like this still needs to be addressed

Reply:
yea it seems i made change in another branch